### PR TITLE
refactor(ied/da-container,do-container): code improvements

### PR DESCRIPTION
**Pull request by JakobVogelsang**
@Flurb  as you did not like the toggle function I tried to refactor. What do you think.

I do get snapshot updates that I cannot interpret. Maybe you can have a look.

### CODE REVIEW

**Comment by Flurb** [comment]
Way better! Using standard components instead of your own implementation is always better.
I couldn't find one, but I think I might have searched wrong, mwc-icon-button-toggle is perfect here.

**Review by Flurb — CHANGES_REQUESTED** [review]
Looks good, but during testing is see that the icon isn't changing when toggling.

**Review by Flurb — CHANGES_REQUESTED** [review]
Also, the same toggle principle can be added to the ln-container :)

**Comment by JakobVogelsang** [comment]
> Also, the same toggle principle can be added to the ln-container :)

This was on purpose from my side (see #461). My opinion is that we need some other mechanism here. Do you agree? Do I miss something?

**Comment by JakobVogelsang** [comment]
> Way better! Using standard components instead of your own implementation is always better. I couldn't find one, but I think I might have searched wrong, mwc-icon-button-toggle is perfect here.

This is bug of this component that only is present in debug mode. Just for you to compare, see the buttons in the log.

**Comment by Flurb** [comment]
> > Way better! Using standard components instead of your own implementation is always better. I couldn't find one, but I think I might have searched wrong, mwc-icon-button-toggle is perfect here.
> 
> This is bug of this component that only is present in debug mode. Just for you to compare, see the buttons in the log.

Then it's fine, hope it will be fixed soon!

**Comment by Flurb** [comment]
> > Also, the same toggle principle can be added to the ln-container :)
> 
> This was on purpose from my side (see #461). My opinion is that we need some other mechanism here. Do you agree? Do I miss something?

Why would this functionality be different for a LN container compared to DO/DA containers?

**Comment by JakobVogelsang** [comment]
> > > Also, the same toggle principle can be added to the ln-container :)
> > 
> > 
> > This was on purpose from my side (see #461). My opinion is that we need some other mechanism here. Do you agree? Do I miss something?
> 
> Why would this functionality be different for a LN container compared to DO/DA containers?

Because rendering the data model for a complicated IED does take some seconds which in my opinion is not a good user experience. The `hidden` attribute is a performance problem, but as is, we still walk through every DO and DA there is in a data model of an IED. We don't need to do that. I think we should on first render only consider the data model up until the logical nodes and only on toggle button trigger should dive into the DOs and DAs. :)

**Review by Flurb — APPROVED** [review]

**Review by Flurb — CHANGES_REQUESTED** [review]
I'm going to implement a solution for all three the containers, should not be a problem :)

**Comment by Flurb** [comment]
@JakobVogelsang (or anyone else) please review :)

File: web-test-runner.config.mjs [hunk 1]
```diff
@@ -12,7 +12,7 @@ export default /** @type {import("@web/test-runner").TestRunnerConfig} */ ({
    * Plugins have a fix URL and do not fit to the file structure in test environment.
    * Creating open-scd in the tests leads to error in the browser log - we had to disable the browser log
   */
-  browserLogs: false,
+  browserLogs: true,
```
Comment (JakobVogelsang) [inline web-test-runner.config.mjs:15]: Just out of curiosity. Why this? Do we need it for something? Due to our plugin architecture, the browser log shows a misleading error when calling `Hosting`. This makes the log hard to read. But if you think we require it for something, I am fine with it.

Comment (Flurb) [reply web-test-runner.config.mjs:15]: No we don't, I just like to use console.logs during testing. For this we need to set it to true, and it should be reverted which I didn't do :)

**Comment by JakobVogelsang** [comment]
LGTM @Flurb

**Review by Flurb — APPROVED** [review]

**Pull request merged**